plugin/store/store.go

```diff
@@ -98,12 +98,31 @@ func (ps *Store) SetState(p *v2.Plugin, state bool) {
 }
 
 // Add adds a plugin to memory and plugindb.
-func (ps *Store) Add(p *v2.Plugin) {
+// An error will be returned if there is a collision.
```

Comment:
Thanks @anusha-ragunathan. I created a PR #28949 for the follow up of using `GetByName()`. Please take a look and let me know if there are any issues.